lib/evm_database_ops.rb

```diff
@@ -119,15 +117,40 @@ def self.restore(db_opts, connect_opts = {})
         backup_folder = action == :dump ? "db_dump" : "db_backup"
         uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
       end
+    else
+      uri = db_opts[:local_file]
+
+      # HACK(ish): Since everything is flowing through MiqFileStorage so that
+      # we can handle the file splitting logic all in one place, this means
+      # that local dumps/backups also need to now have a "uri" in the
+      # connection opts so that MiqLocalMountSession will be used.
```

Comment:
I know I asked for this comment, but I think just the second paragraph gets the point across.

Reply:
Sometimes I write books, and sometimes it is just "word vomit"... you got the latter here.